google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -1758,6 +1763,103 @@ public ApiFuture<QuerySnapshot> get() {
     return get(null);
   }
 
+  ApiFuture<QueryProfile<QuerySnapshot>> getQueryProfileInfo(QueryMode queryMode) {
+    final SettableApiFuture<QueryProfile<QuerySnapshot>> result = SettableApiFuture.create();
```

Comment:
I think we should be more consistent in our use of `final`. My opinion is that it is not required on local variables. The compiler can infer `final` in contexts where it matters (lambdas).

That said, I think `final` on properties, methods, and classes are wonderful since it helps compiler optimize code and actually run faster.

This is more of a `nit` since it makes difference on local properties (other than protect you from accidental assignment).

To reduce line length, I suggest we just omit `final` on all local variables.